inception-recommendation/src/main/java/de/tudarmstadt/ukp/inception/recommendation/project/EvaluationSimulationPanel.java

```diff
@@ -0,0 +1,300 @@
+                chartType.append("'");
+                chartType.append(recommenderName);
+                chartType.append("': 'step', ");
+                dataColumns.append(recommenderName);
+
+                // append data columns
+                dataColumns.append("', ");
+                dataColumns.append(data);
+                dataColumns.append("]");
+                dataColumns.append(",");
+
+                try {
+                    String javascript = createJSScript(dataColumns.toString(),
+                            chartType.toString(), xaxis.toString());
+                    log.debug("Rendering Recommender Evaluation Chart: {}", javascript);
+
+                    aTarget.prependJavaScript(javascript);
+                }
+                catch (IOException e) {
+                    log.error("Unable to render chart", e);
+                    error("Unable to render chart: " + e.getMessage());
+                    aTarget.addChildren(getPage(), IFeedback.class);
+                } 
+            }
+        });
+    }
+
+    @Override
+    public void renderHead(IHeaderResponse aResponse)
+    {
+        super.renderHead(aResponse);
+
+        // import Js
+        aResponse.render(JavaScriptHeaderItem
+                .forReference(new WebjarsJavaScriptResourceReference("c3/current/c3.js")));
+        aResponse.render(JavaScriptHeaderItem
+                .forReference(new WebjarsJavaScriptResourceReference("d3js/current/d3.js")));
+
+        // import Css
+        aResponse.render(
+                CssHeaderItem.forReference(new WebjarsCssResourceReference("c3/current/c3.css")));
+
+    }
+
+    /**
+     * Creates the JS script to render graph with the help of given data points. Also creates an
+     * x-axis of a sequence from 0 to maximumNumberOfPoints (50). Example value of aDataColumns:
+     * 
+     * <pre>
+     * ['recommender1', 1.0, 2.0, 3.0 ], ['recommender2', 2.0, 3.0, 4.0]
+     * </pre>
+     * 
+     * Example value of aChartType
+     * 
+     * <pre>
+     * recommender1: 'step', recommender2 : 'step'
+     * </pre>
+     */
+    private String createJSScript(String aDataColumns, String aChartType, String aXaxis)
+        throws IOException
+    {
+        String xaxisValues = "[ 'x' ," + aXaxis + "]";
+        String data = toJsonString(aDataColumns).substring(1, aDataColumns.toString().length());
+
+        // bind data to chart container
+        String javascript = "var chart=c3.generate({bindto:'#" + chartContainer.getMarkupId()
+                + "',data:{ x:'x', columns:[" + xaxisValues + " ," + data + "],types:{" + aChartType
+                + "}},axis: { y : { tick : { format: function(d){return Math.round(d * 10000) / 10000}}}}});;";
+        return javascript;
+    }
+
+    /**
+     * returns the recommender Engine for the given recommender model
+     * 
+     */
+    private RecommendationEngine getRecommendationEngine(IModel<Recommender> recommenderModel)
```

Comment:
Inject a `RecommenderFactoryRegistry` and use it to obtain via the `RecommendationEngineFactory` required for the selected recommender. The factory is then used to instantiate the actual `RecommendationEngine`s such as `OpenNlpNerRecommender`. Check out uses of the `RecommenderFactoryRegistry` and `RecommendationEngineFactory` elsewhere in the code to see how they are used in practice.

Reply:
Fixed.